daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -1073,6 +1080,13 @@ convertType env o@(TypeCon t ts)
         if supportsArrowType (envLfVersion env) || length ts' == 2
           then foldl TApp TArrow <$> traverse (convertType env) ts'
           else unsupported "Partial application of (->)" o
+    | Just m <- nameModule_maybe (getName t)
+    , GHC.moduleName m == mkModuleName "DA.Internal.LF"
+    , getOccString t == "Pair"
+    , [StrLitTy f1, StrLitTy f2, t1, t2] <- ts = do
```

Comment:
I wonder if we should throw an error if `Pair` is not applied to `StrLitTy` instead of falling into the `otherwise` clause. I don’t think it should matter since users cannot produce use `Pair` anyway but it might make the intention of the code a bit clearer.

Reply:
We could, but it's more code and makes it harder to read.